plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/listeners/CodeWhispererPopupIntelliSenseAcceptListener.kt

```diff
@@ -0,0 +1,52 @@
+// Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codewhisperer.popup.listeners
+
+import com.intellij.codeInsight.lookup.Lookup
+import com.intellij.codeInsight.lookup.LookupEvent
+import com.intellij.codeInsight.lookup.LookupListener
+import com.intellij.codeInsight.lookup.LookupManagerListener
+import com.intellij.codeInsight.lookup.impl.LookupImpl
+import software.aws.toolkits.jetbrains.services.codewhisperer.model.InvocationContext
+import software.aws.toolkits.jetbrains.services.codewhisperer.popup.CodeWhispererPopupManager
+import software.aws.toolkits.jetbrains.services.codewhisperer.service.CodeWhispererInvocationStatus
+
+class CodeWhispererPopupIntelliSenseAcceptListener(private val states: InvocationContext) : LookupManagerListener {
+    override fun activeLookupChanged(oldLookup: Lookup?, newLookup: Lookup?) {
+        if (oldLookup != null || newLookup == null) return
+
+        addIntelliSenseAcceptListener(newLookup, states)
+    }
+}
+
+fun addIntelliSenseAcceptListener(lookup: Lookup, states: InvocationContext) {
+    lookup.addLookupListener(object : LookupListener {
```

Comment:
looks like we can reuse this LookupListener instance, maybe make it an singleton `Object`? Asking because i'm not sure if re-initialization of this listener will add memory burden

Reply:
it it's not an concern then it's fine

Reply:
technically yes but a listener doesn't really add much memory